### ISSUE TITLE
Annotations missing for inline components when annotate_rendered_view_with_filenames is enabled

### ISSUE THREAD

**Issue by dmarcoux** [ViewComponent/view_component#1301]
I am using view_component in a Rails application with `config.action_view.annotate_rendered_view_with_filenames` set to `true`. For view components with templates, the HTML is annotated with the usual `BEGIN` and `END`. However for inline components (so relying on `call` and not having a template), the HTML doesn't contain those annotations.

### Steps to reproduce

1. Set `config.action_view.annotate_rendered_view_with_filenames` set to `true` in your Rails config.
2. Render a view component with a template.
3. Render an inline view component.
4. See the `BEGIN` and `END` comments for the view component with a template, but not for the inline view component.   

Example in which the `<abbr>` tag is an inline view component and another component with a template which renders `Hello!`:
![example](https://user-images.githubusercontent.com/1102934/157336703-1b849559-0f06-42b0-b2ed-c8a59fa4503a.png)

### Expected behavior
I expect the annotations to be added for inline view components.

### Actual behavior
The annotations are not added for inline view components.

### System configuration
**Rails version**: Rails 7.0.2.2

**Ruby version**: ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-linux]

**Gem version**: 2.49.1

**Comment by joelhawksley** [comment]
@dmarcoux thanks for filing this issue! The current implementation of template annotations hooks into the template compilation process, which isn't used for inline components.

I'd love to see this feature added to the framework, would you like to take it on? I'd be happy to pair with you to get it off the ground: joelhawksley@github.com

**Comment by joevandyk** [comment]
would love to have this - would this be done via output_postamble and output_preamble?

**Comment by joelhawksley** [comment]
@joevandyk yes, I think we'd do it in that area of the codebase! Happy to pair with you if you'd like, email me with your availability if you'd like: joelhawksley@github.com

**Comment by joelhawksley** [comment]
I've verified that this issue is no longer present in the current v3 release.

**Comment by joevandyk** [comment]
@joelhawksley  odd. I'm seeing annotations in 3.14.0 as expected, they are missing in 3.15.0.

we're using haml if that matters.

**Comment by joelhawksley** [comment]
@joevandyk my apologies for closing this- my verification was only that the correct `identifier` was being exposed on the component, not that it would be rendered by Rails.

It _shouldn't_ matter that you're using `haml`.

However, I remain confused as to how this case _ever_ worked for you, as annotations are only added to compiled ERB templates (link to the erb handler in Rails 7.1.4.2): https://github.com/rails/rails/blob/v7.1.4.2/actionview/lib/action_view/template/handlers/erb.rb.

I wonder if I'm still missing something about your specific use case. Would you be willing to write a PR with a test that passes on v3.14.0 and fails on v3.15.0?

I could see Rails adding support for template annotations to renderables themselves, or we could add them to the output in ViewComponent. I would guess that the latter is more viable as the ViewComponent is effectively serving as the compiled template method in the case of an inline template.

**Comment by joelhawksley** [comment]
Closed by https://github.com/ViewComponent/view_component/pull/2301
